app/src/main/java/org/mozilla/fenix/SearchWidgetProvider.kt

```diff
@@ -56,11 +62,12 @@ class SearchWidgetProvider : AppWidgetProvider() {
         else -> SearchWidgetProviderSize.EXTRA_SMALL
     }
 
-    private fun getLayout(size: SearchWidgetProviderSize) = when (size) {
+    private fun getLayout(size: SearchWidgetProviderSize, showMic: Boolean) = when (size) {
         SearchWidgetProviderSize.EXTRA_LARGE -> R.layout.search_widget_extra_large
         SearchWidgetProviderSize.LARGE -> R.layout.search_widget_large
         SearchWidgetProviderSize.MEDIUM -> R.layout.search_widget_medium
-        SearchWidgetProviderSize.SMALL -> R.layout.search_widget_small
+        SearchWidgetProviderSize.SMALL -> if (showMic) R.layout.search_widget_small
```

Comment:
Is this logic needed for any other widget sizes or just SMALL?

Reply:
It's just used for SMALL. Is there a better way to handle this?